src/code_d0a70_len_4fe0.c

```diff
@@ -14,7 +14,14 @@ INCLUDE_ASM(s32, "code_d0a70_len_4fe0", func_8013A6E8);
 
 INCLUDE_ASM(s32, "code_d0a70_len_4fe0", func_8013A704);
 
-INCLUDE_ASM(s32, "code_d0a70_len_4fe0", func_8013A854);
+void func_8013A854(u32 i) {
+    UnkStruct8013A854* unkStruct80156954ListPtr;
```

Comment:
Is this required to match? Can you not do this:
```c
if (i < 90) {
    (*D_80156954)[i].unk_14 = 0;
    // ...
}
```
That dereference of D_80156954 will probably become a macro similar to GAME_STATUS once we work out what the struct is.

Reply:
That's a good syntax optimization, I'd like to try and implement it, currently I get this error with these changes.

```
> ./diff.py -mw func_8013A854
cpp -Iinclude -D _LANGUAGE_C -ffreestanding -DF3DEX_GBI_2 src/code_d0a70_len_4fe0.c | tools/cc1 -O2 -quiet -G 0 -mcpu=vr4300 -mfix4300 -mips3 -mgp32 -mfp32 -o - | tools/mips-nintendo-nu64-as -EB -Iinclude -G 0 - -o build/src/code_d0a70_len_4fe0.o
src/code_d0a70_len_4fe0.c: In function `func_8013A854':
src/code_d0a70_len_4fe0.c:37: request for member `unk_14' in something not a structure or union
src/code_d0a70_len_4fe0.c:38: request for member `unk_10' in something not a structure or union
```
![image](https://user-images.githubusercontent.com/49568191/96377253-6e85c280-1139-11eb-82d2-dcffae3839a1.png)

```c
typedef struct {
    /* 0x00 */ char unk_0[0x10];
    /* 0x10 */ s16 unk_10;
    /* 0x14 */ s32 unk_14;
    /* 0x18 */ char unk_18[0x64];
} UnkStruct8013A854 ; // size = 0x7C

typedef UnkStruct8013A854 UnkStruct8013A854List[UNKSTRUCT8013A854_SIZE];

extern UnkStruct8013A854List* D_80156954[UNKSTRUCT8013A854_SIZE];

void func_8013A854(u32 i) {    
    if (i < 90) {
        (*D_80156954)[i].unk_14 = 0;
        (*D_80156954)[i].unk_10 = -1;
    }
}
```

Reply:
@nanaian could I resolve this comment without the optimization or should I try to get this to work?

Reply:
Go ahead and resolve it. Although, I thought `D_80156954` was a pointer to an array, so this _should_ work...

Reply:
Okay no worries, yeah I though so too. I'm not sure if my syntax is off or not, I like your suggestion.